tests/unittests/storage/test_storage.py

```diff
@@ -0,0 +1,363 @@
+from orion.core.worker.trial import Trial
+from orion.storage.base import FailedUpdate, get_storage
+
+storage_backends = [
+    None,  # defaults to legacy with PickleDB
+]
+
+base_experiment = {
+    'name': 'default_name',
+    'version': 0,
+    'metadata': {
+        'user': 'default_user',
+        'user_script': 'abc',
+        'datetime': '2017-11-23T02:00:00'
+    }
+}
+
+
+base_trial = {
+    'experiment': 'default_name',
+    'status': 'new',  # new, reserved, suspended, completed, broken
+    'worker': None,
+    'submit_time': '2017-11-23T02:00:00',
+    'start_time': None,
+    'end_time': None,
+    'heartbeat': None,
+    'results': [
+        {'name': 'loss',
+         'type': 'objective',  # objective, constraint
+         'value': 2}
+    ],
+    'params': [
+        {'name': '/encoding_layer',
+         'type': 'categorical',
+         'value': 'rnn'},
+        {'name': '/decoding_layer',
+         'type': 'categorical',
+         'value': 'lstm_with_attention'}
+    ]
+}
+
+
+def _generate(obj, *args, value):
+    if obj is None:
+        return None
+
+    obj = copy.deepcopy(obj)
+    data = obj
+
+    for arg in args[:-1]:
+        data = data[arg]
+
+    data[args[-1]] = value
+    return obj
+
+
+def make_lost_trial():
+    """Make a lost trial"""
+    obj = copy.deepcopy(base_trial)
+    obj['status'] = 'reserved'
+    obj['heartbeat'] = datetime.datetime.utcnow() - datetime.timedelta(seconds=61 * 2)
+    obj['params'].append({
+        'name': '/index',
+        'type': 'categorical',
+        'value': 'lost_trial'
+    })
+    return obj
+
+
+def generate_trials():
+    """Generate Trials with different configurations"""
+    status = ['completed', 'broken', 'reserved', 'interrupted', 'suspended', 'new']
+    new_trials = [_generate(base_trial, 'status', value=s) for s in status]
+
+    # make each trial unique
+    for i, trial in enumerate(new_trials):
+        trial['params'].append({
+            'name': '/index',
+            'type': 'categorical',
+            'value': i
+        })
+
+    return new_trials
+
+
+def generate_experiments():
+    """Generate a set of experiments"""
+    users = ['a', 'b', 'c']
+    return [_generate(base_trial, 'metadata', 'user', value=u) for u in users]
+
+
+@pytest.mark.parametrize('storage', storage_backends)
+class TestStorage:
+    """Test all storage backend"""
+
+    def test_create_experiment(self, storage):
+        """Test create experiment"""
+        with OrionState(experiments=[], database=storage) as cfg:
+            storage = cfg.storage()
+
+            storage.create_experiment(base_experiment)
+
+            experiments = storage.fetch_experiments({})
+            assert len(experiments) == 1, 'Only one experiment in the database'
+
+            experiment = experiments[0]
+            assert base_experiment == experiment, 'Local experiment and DB should match'
+
+    def test_fetch_experiments(self, storage, name='default_name', user='default_user'):
```

Comment:
Should add a test where query fails. There is only one experiment so this test would also pass with `storage.fetch_experiments({})`.

Reply:
I think you missed this one.